Sources/PromotedCore/ErrorHandling/Errors.swift

```diff
@@ -150,7 +148,18 @@ public enum MetricsLoggerError: Error {
   case unexpectedEvent(_ event: Message)
 
   /// Tried to log a User message without userID and logUserID. Non-fatal.
-  case missingUserIDsInUserMessage
+  case missingLogUserIDInUserMessage
+
+  /// Tried to log a LogRequest batch without userID and logUserID. Non-fatal.
+  case missingLogUserIDInLogRequest
+
+  /// Impression without any joinable fields (content ID, insertion ID).
+  /// Non-fatal.
+  case missingJoinableFieldsInImpression
```

Comment:
Snit - `Fields` more specific to `IDs` or `Keys`.

Reply:
Good point. Done.